src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,46 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final String DATE_FORMAT = "dd.MM.yyyy";
+    private static final String INFO_SEPARATOR = " - ";
+    private static final String DATA_SEPARATOR = " ";
+    private static final int INDEX_OF_DATA = 0;
+    private static final int INDEX_OF_NAME = 1;
+    private static final int INDEX_OF_WORK_HOURS = 2;
+    private static final int INDEX_SALARY_PER_HOUR = 3;
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_FORMAT);
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate startDate = parseDateWithFormatter(dateFrom);
+        LocalDate endDate = parseDateWithFormatter(dateTo);
+        StringBuilder employeeInfo = new StringBuilder("Report for period ")
+                .append(dateFrom)
+                .append(INFO_SEPARATOR)
+                .append(dateTo);
+        for (String name : names) {
+            int money = 0;
+            employeeInfo.append(System.lineSeparator())
+                    .append(name)
+                    .append(INFO_SEPARATOR);
+            for (String currentData : data) {
+                String[] dateSplit = currentData.split(DATA_SEPARATOR);
+                LocalDate date = parseDateWithFormatter(dateSplit[INDEX_OF_DATA]);
+                if ((date.equals(startDate) || date.isAfter(startDate))
+                        && (date.equals(endDate) || date.isBefore(endDate))
+                        && name.equals(dateSplit[INDEX_OF_NAME])) {
+                    money += Integer.parseInt(dateSplit[INDEX_OF_WORK_HOURS])
+                            * Integer.parseInt(dateSplit[INDEX_SALARY_PER_HOUR]);
+                }
+            }
+            employeeInfo.append(money);
```

Comment:
The nested loop structure seems necessary for this task, as you are iterating over names and data. However, make sure there are no redundant loops.